### DISCUSSION TITLE
v2.1.0

### DISCUSSION THREAD

**Discussion by bdbch** [ueberdosis/tiptap#4331] [Releases]
We finally made it to 2.1.0 – here is the list of changes included in this release:

## Minor breaking changes

* Require file extensions for imports and exports by @benasher44 in https://github.com/ueberdosis/tiptap/pull/4001
* update strikethrough shortcut to be inline with other editors by @bdbch in https://github.com/ueberdosis/tiptap/pull/4288

### @tiptap/core
* New Cut Command & Position Mapping for insertContentAt by @bdbch in https://github.com/ueberdosis/tiptap/pull/4141
* Added new options for nodeInputRules for block replacements by @bdbch in https://github.com/ueberdosis/tiptap/pull/4289

### @tiptap/extension-collaboration-cursor
* Add support for custom selection builders by @benglewis in https://github.com/ueberdosis/tiptap/pull/4128

### @tiptap/extension-list-keymap
* New list keymap extension with improved list key handling by @bdbch in https://github.com/ueberdosis/tiptap/pull/3819
* chore(lists): move list keymap to extra extension by @bdbch in https://github.com/ueberdosis/tiptap/pull/4290

### @tiptap/extension-typography
* feat(typography): allow overrides of inputRule outputs by @bdbch in https://github.com/ueberdosis/tiptap/pull/4287

### @tiptap/react
* React context implementation for Tiptap by @bdbch in https://github.com/ueberdosis/tiptap/pull/4192

### @tiptap/core
* Fix empty nodes not behaving the same as empty nodes on document node inside isolating nodes by @bdbch in https://github.com/ueberdosis/tiptap/pull/3943
* Fix `configure` loosing the existing config on Marks and Extensions by @nholden in https://github.com/ueberdosis/tiptap/pull/3822
* Fix 'enter' behavior on android custom node views by @C-Hess in https://github.com/ueberdosis/tiptap/pull/4153
* Fix nested chain not preserving dispatch state by @C-Hess in https://github.com/ueberdosis/tiptap/pull/4152
* Fix text style to be clearable on new lines by @C-Hess in https://github.com/ueberdosis/tiptap/pull/4151
* Fix cut and insertContentAt functions by @bdbch in https://github.com/ueberdosis/tiptap/pull/4187
* Fix classes being duplicated on mergeAttributes function by @bdbch in https://github.com/ueberdosis/tiptap/pull/4320

### @tiptap/extension-strikethrough
* update strikethrough shortcut to be inline with other editors by @bdbch in https://github.com/ueberdosis/tiptap/pull/4288

### @tiptap/extension-list-item
* Fix nested lists style issue by @tommy-typos in https://github.com/ueberdosis/tiptap/pull/4172

### @tiptap/extension-list-keymap
* Fix List issues & add support for Mod keys by @bdbch in https://github.com/ueberdosis/tiptap/pull/4210

### @tiptap/extension-link
* Fix left click invalid by @HengCC in https://github.com/ueberdosis/tiptap/pull/3917
* Fix link not being kept when pasting url with link as text by @bdbch in https://github.com/ueberdosis/tiptap/pull/3975
* Fix links autolinking when not needed by @bdbch in https://github.com/ueberdosis/tiptap/pull/3989
* Fix onPaste handling in link extension in rc.4 by @svenadlung in https://github.com/ueberdosis/tiptap/pull/4072
* Fix Prevent auto-linking when typing URL inside inline code mark by @rfgamaral in https://github.com/ueberdosis/tiptap/pull/4160
* Fixed rel not being able to be set by @lekoala in https://github.com/ueberdosis/tiptap/pull/4112
* Fixes link going to wrong url by @joenarus in https://github.com/ueberdosis/tiptap/pull/4078
* Do not use window.open for links in readonly mode by @thecodrr in https://github.com/ueberdosis/tiptap/pull/4073
* Fix autolinking and pasting by @bdbch in https://github.com/ueberdosis/tiptap/pull/4292

### @tiptap/extension-horizontal-rule
* HorizontalRule: fix setHorizontalRule putting cursor at non-text positions by @bdbch in https://github.com/ueberdosis/tiptap/pull/4319

### @tiptap/extension-bubble-menu
* Fix menu debounce not working with collab by @bdbch in https://github.com/ueberdosis/tiptap/pull/3956

### @tiptap/extension-history
* Fix mod+z/mod+y failed with capslock by @Slapbox in https://github.com/ueberdosis/tiptap/pull/4132

### @tiptap/react
* Fixed missing select state when text selection is around by @bdbch in https://github.com/ueberdosis/tiptap/pull/4148
* Fixed checking props.clientRect before creating ReactRenderer by @HariAcharya80 in https://github.com/ueberdosis/tiptap/pull/4138
* Destroy editor in safe by @KentoMoriwaki in https://github.com/ueberdosis/tiptap/pull/4000

### @tiptap/vue3
* Fix component names in Vue 3 devtools by @segevfiner in https://github.com/ueberdosis/tiptap/pull/3970

### @tiptap/vue2
* Make Vue 2 `nodeViewProps` compatible with `@vue/composition-api` by @FloEdelmann in https://github.com/ueberdosis/tiptap/pull/4123

### Core
* New class on editor wrapper by @patrickbaber in https://github.com/ueberdosis/tiptap/pull/4068
* Require file extensions for imports and exports by @benasher44 in https://github.com/ueberdosis/tiptap/pull/4001
  * (this should now allow support for more node environments)

### Maintainment
* chore: added CODEOWNERS file by @bdbch in https://github.com/ueberdosis/tiptap/pull/4154
* update peerDependencies to fix lerna version tasks by @bdbch in https://github.com/ueberdosis/tiptap/pull/3914

### Docs
* Fix typo in commands.md by @marcuslyons in https://github.com/ueberdosis/tiptap/pull/4038
* Add missing editor getters by @muttenzer in https://github.com/ueberdosis/tiptap/pull/4150
* Update markInputRule.ts by @mcshaz in https://github.com/ueberdosis/tiptap/pull/4162
* Docs/installation cdn by @RicoTrevisan in https://github.com/ueberdosis/tiptap/pull/4045
* Fix typo in guide by @tsukinose81 in https://github.com/ueberdosis/tiptap/pull/4205
* Fix HocusPocus typo in introduction.md by @robintindale in https://github.com/ueberdosis/tiptap/pull/4275
* update cloud docs by @patrickbaber in https://github.com/ueberdosis/tiptap/pull/4247
* update vanilla js docs by @bdbch in https://github.com/ueberdosis/tiptap/pull/4316

### Demos
* updated @hocuspocus/provider to 2.0.1 by @janthurau in https://github.com/ueberdosis/tiptap/pull/3913
* fix demo for cut command by @bdbch in https://github.com/ueberdosis/tiptap/pull/4313

### Build
* build(deps): bump cypress-io/github-action from 5.5.1 to 5.6.1 by @dependabot in https://github.com/ueberdosis/tiptap/pull/3933
* build(deps): bump cypress-io/github-action from 5.6.1 to 5.8.3 by @dependabot in https://github.com/ueberdosis/tiptap/pull/4139
* build(deps): bump actions/checkout from 3.5.0 to 3.5.3 by @dependabot in https://github.com/ueberdosis/tiptap/pull/4122
* build(deps-dev): bump vite from 2.9.15 to 2.9.16 by @dependabot in https://github.com/ueberdosis/tiptap/pull/4110
* build(deps-dev): bump vite from 2.9.13 to 2.9.16 in /demos by @dependabot in https://github.com/ueberdosis/tiptap/pull/4109
* build(deps): bump actions/setup-node from 3.6.0 to 3.7.0 by @dependabot in https://github.com/ueberdosis/tiptap/pull/4196
* build(deps-dev): bump word-wrap from 1.2.3 to 1.2.4 by @dependabot in https://github.com/ueberdosis/tiptap/pull/4232

## New Contributors
* @HengCC made their first contribution in https://github.com/ueberdosis/tiptap/pull/3917
* @nholden made their first contribution in https://github.com/ueberdosis/tiptap/pull/3822
* @muttenzer made their first contribution in https://github.com/ueberdosis/tiptap/pull/4150
* @benasher44 made their first contribution in https://github.com/ueberdosis/tiptap/pull/4001
* @tommy-typos made their first contribution in https://github.com/ueberdosis/tiptap/pull/4172
* @mcshaz made their first contribution in https://github.com/ueberdosis/tiptap/pull/4162
* @HariAcharya80 made their first contribution in https://github.com/ueberdosis/tiptap/pull/4138
* @Slapbox made their first contribution in https://github.com/ueberdosis/tiptap/pull/4132
* @lekoala made their first contribution in https://github.com/ueberdosis/tiptap/pull/4112
* @joenarus made their first contribution in https://github.com/ueberdosis/tiptap/pull/4078
* @segevfiner made their first contribution in https://github.com/ueberdosis/tiptap/pull/3970
* @FloEdelmann made their first contribution in https://github.com/ueberdosis/tiptap/pull/4123
* @tsukinose81 made their first contribution in https://github.com/ueberdosis/tiptap/pull/4205
* @benglewis made their first contribution in https://github.com/ueberdosis/tiptap/pull/4128
* @robintindale made their first contribution in https://github.com/ueberdosis/tiptap/pull/4275

**Full Changelog**: https://github.com/ueberdosis/tiptap/compare/v2.0.4...v2.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ueberdosis/tiptap/releases/tag/v2.1.0'>v2.1.0</a>.</em>

**Comment by nholden** [comment]
👋 Hi, @bdbch! On your first bullet point, you [reverted](https://github.com/ueberdosis/tiptap/pull/3822#issuecomment-1523489076) https://github.com/ueberdosis/tiptap/pull/3822. I took another approach to fix the issue in https://github.com/ueberdosis/tiptap/pull/3998, but I never heard back. Did this issue get resolved in some other way?

**Reply by bdbch** [reply]
You're right, no it wasn't – sorry for that. I'll take a look at your PR again so we can get this in in the next minor release.
